src/HttpClientRequestAdapter.cs

```diff
@@ -52,7 +52,7 @@ public HttpClientRequestAdapter(IAuthenticationProvider authenticationProvider,
             pNodeFactory = parseNodeFactory ?? ParseNodeFactoryRegistry.DefaultInstance;
             sWriterFactory = serializationWriterFactory ?? SerializationWriterFactoryRegistry.DefaultInstance;
             obsOptions = observabilityOptions ?? new ObservabilityOptions();
-            activitySource = new(obsOptions.TracerInstrumentationName);
+            activitySource = ActivitySourceRegistry.DefaultInstance.GetOrCreateActivitySource(obsOptions.TracerInstrumentationName);

```

Comment:
This only exists for net5 and higher.
I think we'll need to rely on conditional compilation, and properly dispose of the instance ourselves for anything lower
https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.activitysource?view=net-8.0

Reply:
If you mean the `ActivitySource` type here, I think this should still be okay for netstandard as the type is brought in using the explicit package reference in the csproj file. There's also won't be a need to dispose really as the registry assures only one instance for the entire app lifetime.

Reply:
I meant the DefaultInstance singleton on the ActivitySourceRegistry

Reply:
I believe the registry works on framework from the tests [here](https://github.com/microsoft/kiota-http-dotnet/blob/40262ef8cc3268af4949db4062d447820eeca404/Microsoft.Kiota.Http.HttpClientLibrary.Tests/Middleware/ActivitySourceRegistryTests.cs#L20). As the [`ActivitySourceRegistry`](https://github.com/microsoft/kiota-http-dotnet/blob/40262ef8cc3268af4949db4062d447820eeca404/src/Middleware/ActivitySourceRegistry.cs#L14) is a custom class in this project, shouldn't the singleton still work without issue? (I suspect there's something I'm still missing here 😅)

Reply:
thanks for clarifying, I got mixed up when searching for the type in the docs... sorry about the confusion.